packages/strapi-admin/admin/src/components/LeftMenuLinkContainer/index.js

```diff
@@ -40,109 +37,70 @@ function LeftMenuLinkContainer({ plugins, ...rest }) {
     return acc;
   }, {});
 
-  const linkSections = Object.keys(pluginsSections).map((current, j) => {
-    const contentTypes = pluginsSections[current].links;
-
-    return (
-      <div key={j}>
-        <p className="title">
-          <FormattedMessage {...messages.contentTypes}>
-            {title => title}
-          </FormattedMessage>
-        </p>
-        <ul className="list  models-list">
-          {sortBy(contentTypes, 'label').map((link, i) => (
-            <LeftMenuLink
-              {...rest}
-              key={`${i}-${link.label}`}
-              icon={link.icon || 'circle'}
-              label={link.label}
-              destination={`/plugins/${link.plugin}/${link.destination ||
-                link.uid}`}
-            />
-          ))}
-        </ul>
-      </div>
-    );
-  });
-
-  // Check if the plugins list is empty or not and display plugins by name
-  const pluginsLinks = !isEmpty(plugins) ? (
-    map(sortBy(plugins, 'name'), plugin => {
-      const shouldInjectPlugin = !!plugin.mainComponent;
-
-      if (
+  // Generate the list of plugin links
+  const pluginsLinks = Object.values(plugins)
+    .filter(
+      plugin =>
         plugin.id !== 'email' &&
         plugin.id !== 'content-manager' &&
-        shouldInjectPlugin
-      ) {
-        const pluginSuffixUrl = plugin.suffixUrl
-          ? plugin.suffixUrl(plugins)
-          : '';
+        !!plugin.mainComponent
```

Comment:
Can you add a comment here saying that plugins without a mainComponent should not appear in the left menu?